cmd/fs-v1-multipart.go

```diff
@@ -474,7 +605,22 @@ func (fs fsObjects) CompleteMultipartUpload(bucket string, object string, upload
 		return "", err
 	}
 
-	uploadIDPath := path.Join(bucket, object, uploadID)
+	bucketDir, err := fs.getBucketDir(bucket)
+	if err != nil {
+		return "", toObjectErr(err, bucket)
+	}
+
+	if _, err = fsStatDir(bucketDir); err != nil {
+		return "", toObjectErr(traceError(err), bucket)
+	}
+
+	uploadIDPath := pathJoin(bucket, object, uploadID)
+
+	rwlk, err := fs.rwPool.Create(pathJoin(fs.fsPath, minioMetaMultipartBucket, bucket, object, uploadsJSONFile))
+	if err != nil {
+		return "", toObjectErr(traceError(err), bucket, object)
+	}
+	defer rwlk.Close()
```

Comment:
this should be done at the bottom of the function below these lines:
```
        // Hold the lock so that two parallel complete-multipart-uploads                                                                                                        
        // do not leave a stale uploads.json behind.                                                                                                                            
        objectMPartPathLock := globalNSMutex.NewNSLock(minioMetaMultipartBucket,                                                                                                
                pathJoin(bucket, object))                                                                                                                                       
        objectMPartPathLock.Lock()                                                                                                                                              
        defer objectMPartPathLock.Unlock() 
```